pkg/brownfield/certificates.go

```diff
@@ -0,0 +1,110 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+package brownfield
+
+import (
+	"strings"
+
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/golang/glog"
+
+	ptv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressprohibitedtarget/v1"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/utils"
+)
+
+type certsByName map[string]n.ApplicationGatewaySslCertificate
+
+// GetBlacklistedCerts filters the given list of certificates to the list of certs that AGIC is allowed to manage.
+func GetBlacklistedCerts(certificates []n.ApplicationGatewaySslCertificate, allExistingListeners []n.ApplicationGatewayHTTPListener, prohibited []*ptv1.AzureIngressProhibitedTarget) ([]n.ApplicationGatewaySslCertificate, []n.ApplicationGatewaySslCertificate) {
+
+	// Get the list of blacklisted listeners, from which we can determine what certificates should be blacklisted.
+	existingBlacklistedListeners, _ := GetBlacklistedListeners(allExistingListeners, prohibited)
+	blacklistedCertSet := make(map[string]interface{})
```

Comment:
```go
if len(existingBlacklistedListeners) > 0 {
    return nil, certificates
}
```

Then we can remove also remove `prohibited []*ptv1.AzureIngressProhibitedTarget` as param.

Reply:
Great idea!